File: QuickComplaint.Data/PagedResult.vb
Imports System
Imports System.Collections.ObjectModel

Public Class PagedResult (Of T)

    Public Sub New(requestedPage As Integer, requestedPageSize As Integer, recordCount As Int64,
                   data As Collection(Of T))
        Dim totalPages As Int32
        If recordCount > 0 Then
            totalPages = CInt(Math.Ceiling(CDbl(recordCount)/requestedPageSize))
        End If

        TotalCount = recordCount
        PageCount = totalPages
        PageSize = requestedPageSize
        CurrentPage = requestedPage
        Results = data
    End Sub

    Public Property Results As Collection(Of T)
    Public Property CurrentPage As Integer
    Public Property PageCount As Integer
    Public Property PageSize As Integer
    Public Property TotalCount As Int64
End Class

File: QuickComplaint.Data/PagedResult.cs
using System;
using System.Collections.ObjectModel;

public class PagedResult<T>
{

	public PagedResult(int requestedPage, int requestedPageSize, Int64 recordCount, Collection<T> data)
	{
		Int32 totalPages = default(Int32);
		if (recordCount > 0) {
			totalPages = Convert.ToInt32(Math.Ceiling(Convert.ToDouble(recordCount) / requestedPageSize));
		}

		TotalCount = recordCount;
		PageCount = totalPages;
		PageSize = requestedPageSize;
		CurrentPage = requestedPage;
		Results = data;
	}

	public Collection<T> Results { get; set; }
	public int CurrentPage { get; set; }
	public int PageCount { get; set; }
	public int PageSize { get; set; }
	public Int64 TotalCount { get; set; }
}
